src/Shared/DefaultSerializationContext.cs

```diff
@@ -16,17 +16,142 @@
 
 #endregion
 
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
 using Grpc.Core;
 
 namespace Grpc.Shared
 {
     internal sealed class DefaultSerializationContext : SerializationContext
     {
-        public byte[]? Payload { get; set; }
```

Comment:
The logic of DefaultSerializationContext looks very similar to HttpContextSerializationContext. It would be nice if we could somehow deduplicate.

Reply:
I tried to come up with a shared context but `HttpContextSerializationContext` is deeply integrated with `HttpContext`, and that type doesn't exist in the client.